### DISCUSSION TITLE
How to add screenshot to a specific place

### DISCUSSION THREAD

**Discussion by ShifferB** [allure-framework/allure2#2097] [Questions & Support] [answered]
Hi there! 
I need to attach a screenshot in a specific place. so it should be not after the test passed but inside a test execution

**Comment by PavelMal** [comment, marked as answer]
Hello @ShifferB 

You can implement it in several ways:
1. Allure.addAttachment(); 
example:
```
public static void check(String someData) {
    Allure.addAttachment("Data", someData);
    assertEquals("data", someData);
}
```
 need to add dependency to maven/gradle: 
```
<dependency>
    <groupId>io.qameta.allure</groupId>
    <artifactId>allure-java-commons</artifactId>
    <version>$LATEST_VERSION</version>
</dependency>
```

2. @Attachment (Used to mark methods that produce attachments. Returned value of such methods will be copied and shown in the report as attachment.)
example:
```
@Attachment
@Step("Get some data")
public String getData() {
    return "Hello World";
}
```

need to add the same ‘allure-java-commons’ dependency

**Reply by ShifferB** [reply]
Much appreciation, @PavelMal!
I managed to resolve it utilizing the first option.

**Reply by kiragengis** [reply]
How can I achieve this with C# and NUnit?

**Reply by baev** [reply]
Wrap attachment with the step with the same name:

```c#
AllureApi.Step("attachment name", () => {
    AllureApi.AddAttachment("attachment name", ...)
})
```
